src/core/cpu_profiler.cc

```diff
@@ -134,14 +142,15 @@ void cpu_profiler::on_signal() {
     // Note: this only protects against C++ exceptions, therefore foreign
     // exceptions or long jumps could still cause segfaults within the profiler.
     const bool no_uncaught_exceptions = std::uncaught_exceptions() == 0;
-    if(!no_uncaught_exceptions) {
+
+    if (force_drop_stacktraces) {
+        _stats.dropped_samples_from_manual_disablement++;
+    } else if (!no_uncaught_exceptions) {
         _stats.dropped_samples_from_exceptions++;
-    }
+    } else if (auto guard_opt = _traces_mutex.try_lock(); guard_opt.has_value()) {
+        // Skip the sample if the main thread is currently reading
+        // _traces. This case shouldn't happen often though.
```

Comment:
This fixes a small accounting bug, where we would double count drops from exceptions as dropped from exceptions and from lock contention.